core/broadcaster.go

```diff
@@ -0,0 +1,106 @@
+// Package core/broadcast implements multi-listener broadcast channels.
```

Comment:
Left this in core. Should it be its own package?

Reply:
cool stuff. yeah, seems like it could be it's own package